src/dcmp/dcmp.c

```diff
@@ -1871,11 +1892,30 @@ static int dcmp_output_flist_match(
         int ret = dcmp_strmap_item_index(map, key, &idx);
         assert(ret == 0);
 
-        if (dcmp_disjunction_match(output->disjunction, map, key)) {
+        if (dcmp_disjunction_match(output->disjunction, map, key, is_src)) {
             (*number)++;
             mfu_flist_file_copy(flist, idx, new_flist);
         }
     }
+
+    list_for_each_entry(conjunction,
+                        &output->disjunction->conjunctions,
+                        linkage) {
+        if (is_src) {
+            uint64_t src_matched_tmp;          /* matched src items in this conjunction */
+            src_matched_tmp = conjunction->src_matched;          /* matched src items in this conjunction */
+
+            MPI_Allreduce(&src_matched_tmp, &(conjunction->src_matched), 1,
+                          MPI_UINT64_T, MPI_SUM, MPI_COMM_WORLD);
+        }
+        else {
+            uint64_t dst_matched_tmp;          /* matched dst items in this conjunction */
+            dst_matched_tmp = conjunction->dst_matched;          /* matched dst items in this conjunction */
+
+            MPI_Allreduce(&dst_matched_tmp, &(conjunction->dst_matched), 1,
+                          MPI_UINT64_T, MPI_SUM, MPI_COMM_WORLD);
+        }
+    }
     return 0;
```

Comment:
@KnightKu If you use an flist for src and dst matched, then we can remove this MPI code in the tool. That is how we also keep track of our src_compare and dst_compare lists for syncing directories (in dcmp_strmap_compare in dcmp.c for an example). If you add the matched items to the flist, then you can do a "mfu_flist_global_size." I think this would be simpler.

Reply:
@KnightKu We try to keep MPI calls in the common library, and out of the tools themselves.

Reply:
Hi @dsikich , yeah, agreed, let's follow code conventions, will update it soon, thanks.